Describe PieChart annotations (#1318)
DevExpress/devextreme-documentation: api-reference/20 Data Visualization Widgets/dxPieChart/1 Configuration/annotations/annotations.md

---
---
##### shortDescription
Specifies the annotation collection.

---

> Annotations are containers for images, text blocks and custom content that provide additional information about the visualized data.

Reviewer: ```suggestion
Annotations are containers for images, text blocks, and custom content that provide additional information about the visualized data.
```

                />
                <Annotation
                    type="custom"
                    render={CustomAnnotation}
                />
            </PieChart>
        );
    }

---

Annotations can be unattached or anchored to a chart element. The following list shows how to position them. Chart coordinates ([argument]({basewidgetpath}/Configuration/annotations/#argument) and [series]({basewidgetpath}/Configuration/annotations/#series)) specify the element that the annotation's arrow points to; pixel coordinates ([x]({basewidgetpath}/Configuration/annotations/#x) and [y]({basewidgetpath}/Configuration/annotations/#y)) specify the position of the annotation's center.

- **Unanchored annotation**

        annotations: [{
            x: 100,
            y: 200
        }]

- **Annotation anchored to a series point**

        annotations: [{
            argument: "California",
            series: "States" // required if the PieChart contains more than one series
        }]

- **Annotation positioned at an argument's edge**

        annotations: [{
            argument: "Alaska",
            series: "States",
            location: "edge"
        }]

> When a user long-presses an annotation or hovers the mouse pointer over it, the **PieChart** displays a [tooltip]({basewidgetpath}/Configuration/annotations/#tooltipEnabled).

Reviewer: ```suggestion
The **PieChart** displays a [tooltip]({basewidgetpath}/Configuration/annotations/#tooltipEnabled) when a user long-presses an annotation or hovers the mouse pointer over it.
```

> Objects in the **annotations[]** array configure individual annotations. To specify options common for all annotations, use the [commonAnnotationSettings]({basewidgetpath}/Configuration/commonAnnotationSettings/) object. Individual settings take precedence over common settings.

Reviewer: ```suggestion
Objects in the **annotations[]** array configure individual annotations. To specify options that apply to all annotations, use the [commonAnnotationSettings]({basewidgetpath}/Configuration/commonAnnotationSettings/) object. Individual settings take precedence over common settings.
```

        return (
            <PieChart ... >
                <Annotation
                    type="text"
                    text="Annotation text"
                />
                <Annotation
                    type="image"
                    image="http://image/url/myimage.png"
                />
                <Annotation
                    type="custom"
                    render={CustomAnnotation}
                />
            </PieChart>
        );
    }

---

Annotations can be unattached or anchored to a chart element. The following list shows how to position them. Chart coordinates ([argument]({basewidgetpath}/Configuration/annotations/#argument) and [series]({basewidgetpath}/Configuration/annotations/#series)) specify the element that the annotation's arrow points to; pixel coordinates ([x]({basewidgetpath}/Configuration/annotations/#x) and [y]({basewidgetpath}/Configuration/annotations/#y)) specify the position of the annotation's center.

- **Unanchored annotation**

        annotations: [{
            x: 100,
            y: 200
        }]

- **Annotation anchored to a series point**

        annotations: [{
            argument: "California",

>             // series: "States"
>             // specify this option if the pie chart contains more than one series 

Reviewer 2: ```suggestion
            series: "States" // required if the PieChart contains more than one series
```

        }]

- **Annotation positioned at an argument's edge**

>         annotations: [{
>             argument: "California",
>             series: "States",
>             location: "center"
>         }, {
>             argument: "Alaska",
>             series: "States",
>             location: "edge"
>         }]

Reviewer 2: ```suggestion
        annotations: [{
            argument: "Alaska",
            series: "States",
            location: "edge"
        }]
```

        </DxPieChart>
    </template>

    <script>
    import DxPieChart, {
        DxAnnotation
    } from 'devextreme-vue/pie-chart';

    export default {
        components: {
            DxPieChart,
            DxAnnotation
        },
        data() {
            // ...
        }
    }
    </script>

##### React

    <!-- tab: App.js -->
    import React from 'react';

    import PieChart, {
        Annotation
    } from 'devextreme-react/pie-chart';

    function CustomAnnotation(annotation) {
        const data = annotation.data;
        return (
            <svg>
                {/* Declare custom SVG markup here */}
            </svg>
        );
    }

    export default function App() {
        return (
            <PieChart ... >
                <Annotation
                    type="text"
                    text="Annotation text"
                />
                <Annotation
                    type="image"
                    image="http://image/url/myimage.png"
                />
                <Annotation
                    type="custom"
                    render={CustomAnnotation}
                />
            </PieChart>
        );
    }

---

> Annotations can be anchored to a **PieChart** element. To do this, use the [argument]({basewidgetpath}/Configuration/annotations/#argument) or [series]({basewidgetpath}/Configuration/annotations/#series) options, depending on the element you want to anchor the annotation to.
> 
> To define the position of an unanchored annotation, set the pixel coordinates ([x]({basewidgetpath}/Configuration/annotations/#x) and [y]({basewidgetpath}/Configuration/annotations/#y)) options.

Reviewer 2: The list with examples is detached from the paragraph about anchored elements, although it illustrates how to position them. I suggest taking the text from the [Chart description](http://jsserver.corp.devexpress.com:8080/Documentation/20_2/ApiReference/Data_Visualization_Widgets/dxChart/Configuration/annotations/):

```suggestion
Annotations can be unattached or anchored to a chart element. The following list shows how to position them. Chart coordinates ([argument]({basewidgetpath}/Configuration/annotations/#argument) and [series]({basewidgetpath}/Configuration/annotations/#series)) specify the element that the annotation's arrow points to; pixel coordinates ([x]({basewidgetpath}/Configuration/annotations/#x) and [y]({basewidgetpath}/Configuration/annotations/#y)) specify the position of the annotation's center.
```

Reviewer 2: BTW, the PolarChart's annotations [have the same issue](http://jsserver.corp.devexpress.com:8080/Documentation/20_2/ApiReference/Data_Visualization_Widgets/dxPolarChart/Configuration/annotations/). Let's fix it there too.

- **Unanchored annotation**

        annotations: [{
            x: 100,
            y: 200
        }]

- **Annotation anchored to a series or series point**

        annotations: [{
            argument: "California",

>             series: "States"

Reviewer 2: In most cases, a PieChart contains a single series. Should it always be specified?

        }]

> - **Annotation anchored to an argument's edge or center**

Reviewer 2: Since annotations are positioned at the center by default, I suggest demonstrating only how to position them at the edge.

```suggestion
- **Annotation positioned at an argument's edge**
```

        annotations: [{
            argument: "California",
            series: "States",
            location: "center"
        }, {
            argument: "Alaska",
            series: "States",
            location: "edge"
        }]

> 

Reviewer 2: Is mixed anchoring supported?

Reviewer 3: No, it is not supported
